File: code_samples/dart/design_patterns_adapter.dart
// Design Patterns: Adapter
// AI/ML Training Sample

class Adapter {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = Adapter();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}

File: code_samples/kotlin/design_patterns_adapter.kt
/**
 * Design Patterns: Adapter
 * AI/ML Training Sample
 */

class Adapter {
    var data: String = ""
        private set
    
    fun process(input: String) {
        data = input
    }
    
    fun getData(): String = data
    
    fun validate(): Boolean = data.isNotEmpty()
}

fun main() {
    val instance = Adapter()
    instance.process("example")
    println("Data: ${instance.getData()}")
    println("Valid: ${instance.validate()}")
}
